htdocs/admin/content/alliance/edit.inc.php

```diff
@@ -1,638 +1,534 @@
+	saveDiplomacy($repository);
+} elseif (isset($_POST['res_save']) && $_POST['res_save'] != "") {
+	saveResources($repository, $id);
+} elseif (isset($_POST['buildings']) && $_POST['buildings'] != "") {
+	buildings($buildingRepository, $id);
+} elseif (isset($_POST['techs']) && $_POST['techs'] != "") {
+	technologies($technologyRepository, $id);
+}
+edit($repository, $buildingRepository, $technologyRepository, $id);
+
+function saveInfo(AllianceRepository $repository, int $id)
+{
+	global $twig;
+
+	//  Bild löschen wenn nötig
+	if (isset($_POST['alliance_img_del'])) {
+		$picture = $repository->getPicture($id);
+		if ($picture != null) {
+			if (file_exists('../' . ALLIANCE_IMG_DIR . "/" . $picture)) {
+				unlink('../' . ALLIANCE_IMG_DIR . "/" . $picture);
 			}
-
-			// Daten speichern
-			dbquery("
-			UPDATE
-				alliances
-			SET
-				alliance_name='".$_POST['alliance_name']."',
-				alliance_tag='".$_POST['alliance_tag']."',
-				alliance_text='".addslashes($_POST['alliance_text'])."',
-				alliance_application_template='".addslashes($_POST['alliance_application_template'])."',
-				alliance_url='".$_POST['alliance_url']."',
-				alliance_founder_id='".$_POST['alliance_founder_id']."'
-				".$img_sql."
-			WHERE
-				alliance_id='".$id."'
-			;");
-
-			$twig->addGlobal('successMessage', 'Allianzdaten aktualisiert!');
+			$repository->clearPicture($id);
 		}
-		elseif (isset($_POST['member_save']) && $_POST['member_save']!="")
-		{
-			// Mitgliederänderungen
-			if (isset($_POST['member_kick']) && count($_POST['member_kick'])>0)
-				foreach($_POST['member_kick'] as $k=>$v)
-					dbquery("UPDATE
-						users
-					SET
-						user_alliance_id=0,
-						user_alliance_rank_id=0
-					WHERE
-						user_id='$k';");
-			if (count($_POST['member_rank'])>0)
-				foreach($_POST['member_rank'] as $k=>$v)
-					dbquery("UPDATE
-						users
-					SET
-						user_alliance_rank_id=$v
-					WHERE
-						user_id='$k';");
-			// Ränge speichern
-			if (isset($_POST['rank_del']) && count($_POST['rank_del'])>0)
-				foreach($_POST['rank_del'] as $k=>$v)
-				{
-					dbquery("DELETE FROM alliance_ranks WHERE rank_id='$k';");
-					dbquery("DELETE FROM alliance_rankrights WHERE rr_rank_id='$k';");
-				}
-			if (count($_POST['rank_name'])>0)
-				foreach($_POST['rank_name'] as $k=>$v)
-					dbquery("UPDATE
-						alliance_ranks
-					SET
-						rank_name='".addslashes($v)."',
-						rank_level='".$_POST['rank_level'][$k]."'
-					WHERE
-						rank_id='$k';");
-			$twig->addGlobal('successMessage', 'Mitglieder aktualisiert!');
+	}
+
+	$repository->update($id, [
+		'name' => $_POST['alliance_name'],
+		'tag' => $_POST['alliance_tag'],
+		'text' => $_POST['alliance_text'],
+		'template' => $_POST['alliance_application_template'],
+		'url' => $_POST['alliance_url'],
+		'founder' => $_POST['alliance_founder_id'],
+	]);
+
+	$twig->addGlobal('successMessage', 'Allianzdaten aktualisiert!');
+}
+
+function saveMembers(AllianceRepository $repository)
+{
+	global $twig;
```

Comment:
Instead of using global here. Would pass the property as function argument. Also tells you the type of it

Reply:
Done 4f542f9